### PR TITLE
V3 feature audio - clean up

### LINKED ISSUE #3467: [Audio] Local tracks with special characters in path wont return a value in all cases
# Command bugs

#### Command name

[p]local folder
[p]local play

#### What cog is this command from?
audio

#### What were you expecting to happen?

trying to play a folder with spaces in path such as:
https://puu.sh/F4B4A/9070d830aa.png
and selecting 2

should result in playing all songs inside that folder

#### What actually happened?
Nothing found message.
http://puu.sh/F4B5s/45eb04af34.png

#### How can we reproduce this issue?

make (symlink) folder. add another folder inside it with a space in name. "redbot test" and try to play it.

### LINKED ISSUE #3506: [Audio] inadequate playlist update with local cache enabled
# Other bugs

#### What were you trying to do?

Update local playlist from remote (youtube)

#### What were you expecting to happen?

Update playlist with adding new songs, removal of nonexistent in remote playlist

#### What actually happened?

Random songs was removed, when they are still available on remote playlist.

#### How can we reproduce this issue?

1. use Audio with disabled local cache
2. `playlist save` remote playlist ([i.e.](https://www.youtube.com/playlist?list=PLP08gCf9c_7OITwu6VfPSMkwUsg0RGMDq))
3. enable local cache
4. `playlist update` saved playlist
5. random songs get removed

### PR DESCRIPTION
**Pull request by Drapersniper**
Fixes broken stuff from previous PRs 
Fixes #3467, where paths with special characters no longer return an empty list from `glob.glob`
Attempts to fix #3506 by ensuring that `[p]playlist update` no longer uses the local cache
When selecting a folder with child folders with `[p]localplay` the bot will not enqueue all nested tracks

### CODE REVIEW

**Comment by PredaaA** [comment]
After talks in Discord, I put this as a reminder for you @Drapersniper.
When something is blacklisted in `[p]audioset restrictions`, like Spotify, it should not do any fetch to Spotify API. Not only when trying to enqueue them, but also with `[p]playlist save | append` commands.

Another thing is that `Loading track xx/xx...` message on `[p]playlist save | append` commands get locked when the job is finished.

**Comment by Drapersniper** [comment]
This is not ready to be merged, further work is required, and a full functionality review is required however once this one is proven to be functional
https://github.com/Cog-Creators/Red-DiscordBot/pull/3459 should be ready to be merged into V3/develop.

The "further work required" is ensuring all bugs currently in this branch are resolved.

The way i coded this branch, it contacts all previous work that went into the feature branch so everything can be tested here.

Once this is merged into the feature branch the audio refactor can be considered complete.

**Review by aikaterna — COMMENTED** [review]
Not an exhaustive list. This is what I've found so far - I think you may have already addressed the playlist picker as I see from one of your later commits. 

[p]queue cleanself
Please title case "Removed tracks from the queue" like the rest of the error title messages.

[p]help Audio says `Class joining all Audio subclasses` - make this docstring more user friendly

[p]audioset autoplay reset - possibly display default value in message? (message now is `Set auto-play playlist to default value.`)

[p]audioset countrycode - make the docstring specify Spotify

No matching is present in playlist names across scopes.
Example:
save a playlist as a user scope playlist
play it with [p]playlist start playlistname
playlistname is not matched to any playlist
playlist id is also not matched

Playlist searcher is not seeming to pop up for loading a playlist. Have multiple playlists named the same thing and it's not asking which one to use. Playlist dedupe and playlist remove work appropriately (search popup occurs).

Formatting breaks on <https://www.youtube.com/watch?v=ocDlOD1Hw9k> in user facing responses because of ** in track name

The following commands and/or command groups should not be available to plebs/users without mod-admin perms in the server:
audioset autoplay command group
audioset dc
audioset notify
audioset restrictions command group

"The path must not contain spaces." can be removed from audioset localpath

Used: [p]playlist copy ENM --from-guild 154659242253484033 --to-guild 133049272517001216
Bot response: You cannot use --to-server       <-not embedded, add period at end, format like other "exceptions"

**Comment by Drapersniper** [comment]
# Done
- [p]queue cleanself
- [p]help Audio
- [p]audioset autoplay reset 
- [p]audioset countrycode
- No matching is present in playlist names across scopes.
- Playlist searcher is not seeming to pop up for loading a playlist
- "The path must not contain spaces." can be removed from audioset localpath
- Formatting breaks

# Questions
- You cannot use --to-server <-not embedded, add period at end, format like other "exceptions"
  - This is not handled by Audio, its an early exit in the ArgParser which raises a BadArgument thats handled by the bot Event Handler, should we wish to format this we would need to let Audio handle these which would end up over complicating the core around this area (Errors handled by the Bot Event Handler do not have embed perms like Audio is allowed to enforce).
```py
commands.BadArgument("You cannot use `--guild`")
```
- The following commands and/or command groups should not be available to plebs/users without mod-admin perms in the server:
  - This is already locked behind a check
```py
    @checks.mod_or_permissions(manage_messages=True)
```

File: redbot/cogs/audio/core/utilities/formatting.py [hunk 1]
```diff
@@ -62,7 +65,10 @@ class FormattingUtilities(MixinMeta, metaclass=CompositeMetaClass):
             if search_track_num == 0:
                 search_track_num = 5
             if playlist:
-                name = f"**[{entry.get('name')}]({entry.get('url')})** - {str(entry.get('tracks')) + ' ' + _('tracks')}"
+                name = (
+                    f"**[{entry.get('name')}]({entry.get('url')})**"
+                    f" - {str(entry.get('tracks')) + ' ' + _('tracks')}"
```
Comment (Drapersniper) [inline redbot/cogs/audio/core/utilities/formatting.py]:
```suggestion
                    " - {} {}".format(str(entry.get('tracks')),  _('tracks'))

```

**Review by aikaterna — DISMISSED** [review]

**Review by aikaterna — APPROVED** [review]

**Pull request merged**